subsys/bluetooth/host/gatt.c

```diff
@@ -128,46 +130,95 @@ static int gatt_register(struct bt_gatt_service *svc)
 	return 0;
 }
 
-void bt_gatt_init(void)
-{
-	/* Register mandatory services */
-	gatt_register(&gap_svc);
-	gatt_register(&gatt_svc);
-}
-
-static struct k_sem sc_sem = K_SEM_INITIALIZER(sc_sem, 1, 1);
+static struct gatt_sc {
+	struct bt_gatt_indicate_params params;
+	u16_t start;
+	u16_t end;
+	struct k_sem sem;
```

Comment:
The semaphore seems a bit overkill. At most you'll need a flag to say that the procedure should be redone once the confirmation to a pending indication is received.